### PR TITLE
feat: add pagination decorator docs

### PR DESCRIPTION
**Pull request by dbluhm**
Signed-off-by: Daniel Bluhm <dbluhm@pm.me>

### CODE REVIEW

File: docs/decorators/pagination/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+`results~paginate`. This usage is outlined by message type definitions as
+required.
+
+Example:
+
+```jsonc
+{
+    "...",
+    "~paginate": {
+        "@type": ".../decorators/pagination/paginate",
```
Comment (dbluhm) [inline docs/decorators/pagination/README.md:29]: I'm not sure about the inclusion of `@type` in either decorator.

**Comment by TelegramSam** [comment]
Your proposal works well for offset pagination, but not for other strategies. Here is a decent blog post that expands on the problem. https://slack.engineering/evolving-api-pagination-at-slack/#:~:text=Offsets,to%20return%20results%20for%2C%20page. (ignore the relay cursor section). Their conclusion is a good one, and one that I would recommend here.

File: docs/decorators/pagination/README.md (hunk 1 above)
Comment (TelegramSam) [reply docs/decorators/pagination/README.md:29]: I don't think we need to include the type here, Containing it within a decorator is enough.

File: docs/decorators/pagination/README.md [hunk 2]
```diff
@@ -7,16 +7,26 @@ Pagination Decorators
...
 
-The pagination decorators defines elements common to messages querying for and
-returning a list of results where paging is desirable.
+The pagination decorators define elements common to messages querying for and
+returning a list of results where paging is desirable. These decorators use a
+cursor style paging mechanism for flexibility; for simple collections where
+offsets are sufficient, offset based pagination can be implemented using cursor
+values such as `offset:10` while more complex collections accessed from a
+database or other sources can use cursors representing rows in a table with
+unique identifiers.
```
Comment (TelegramSam) [inline docs/decorators/pagination/README.md:16]: we should include a note that the cursor should be considered an opaque string, and should not be interpreted or manipulated by the querying party.

File: docs/decorators/pagination/README.md [hunk 3]
```diff
@@ -26,16 +36,19 @@ Example:
...
-        "offset": 10
     }
 }
 ```
 
-`limit`: return at most `n` messages.
+`cursor` (optional): Cursor representing where the next page of results begins.
+If omitted, results are returned from the beginning of the collection. Cursor
+corresponds to the returned `next_cursor` of the `~page` decorator denoting page
+information from a paginated collection. The exact value of these cursors is
```
Comment (TelegramSam) [inline docs/decorators/pagination/README.md:48]: not by protocols, but by the parties sending the results.

**Review by burdettadam — APPROVED** [review]

**Pull request merged**